sky/utils/controller_utils.py

```diff
@@ -402,6 +406,14 @@ def shared_controller_vars_to_fill(
         # Only set the SKYPILOT_CONFIG env var if the user has a config file.
         env_vars[
             skypilot_config.ENV_VAR_SKYPILOT_CONFIG] = remote_user_config_path
+    # Check if VPN auth key should be set.
+    if service_config is not None:
+        # Only support Tailscale VPN now
+        if service_config.vpn_config is not None:
+            env_vars.update(service_config.vpn_config.get_setup_env_vars())
+            vpn_config_yaml = yaml.dump(
+                service_config.vpn_config.to_yaml_config())
+            vars_to_fill['vpn_config'] = vpn_config_yaml
```

Comment:
This is not "shared controller env vars". Should we do this in serve.up?

Also in serve.update.